cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -184,19 +185,33 @@ func (s *storageExt) Start(ctx context.Context, host component.Host) error {
 		case cfg.Elasticsearch != nil:
 			esTelset := telset
 			esTelset.Metrics = scopedMetricsFactory(storageName, "elasticsearch", "tracestore")
-			factory, err = es.NewFactory(
-				ctx,
-				*cfg.Elasticsearch,
-				esTelset,
-			)
+			var esAuthCfg *config.AuthExtensionConfig
+			if cfg.Elasticsearch.Authentication.AuthenticatorID != (component.ID{}) {
+				esAuthCfg = &config.AuthExtensionConfig{
+					AuthenticatorID: cfg.Elasticsearch.Authentication.AuthenticatorID,
+				}
+			}
+			httpAuth, authErr := s.resolveAuthenticator(host, esAuthCfg, "elasticsearch", storageName)
```

Comment:
too much boilerplate that can be handled by the helper function

```suggestion
			httpAuth, authErr := s.resolveAuthenticator(host, cfg.Elasticsearch.Authentication, "elasticsearch", storageName)
```